src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -1886,7 +1889,13 @@ internal byte[] SerializeDouble(double v)
                 throw ADP.ParameterValueOutOfRange(v.ToString());
             }
 
+#if NETCOREAPP
+            byte[] bytes = new byte[8];
+            BinaryPrimitives.WriteInt64LittleEndian(bytes, BitConverter.DoubleToInt64Bits(v));
+            return bytes;
```

Comment:
same here, can we avoid the array allocation entirely or at least use the stack?

Reply:
The Function returns an byte[], using stackalloc will free the memory after the return